libretroshare/src/retroshare/rsgxsifacetypes.h

```diff
@@ -140,14 +140,20 @@ struct RsMsgMetaData : RsSerializable
     std::string mMsgName;
     rstime_t      mPublishTs;
 
-    /// the lower 16 bits for service, upper 16 bits for GXS
-    uint32_t    mMsgFlags;
+	/** the lower 16 bits for service, upper 16 bits for GXS
+	 * @todo mixing service level flags and GXS level flag into the same member
+	 * is prone to confusion, use separated members for those things, this could
+	 * be done without breaking network retro-compatibility */
+	uint32_t mMsgFlags;
 
     // BELOW HERE IS LOCAL DATA, THAT IS NOT FROM MSG.
     // normally READ / UNREAD flags. LOCAL Data.
 
-    /// the first 16 bits for service, last 16 for GXS
-    uint32_t    mMsgStatus;
+	/** the first 16 bits for service, last 16 for GXS
+	 * @todo mixing service level flags and GXS level flag into the same member
+	 * is prone to confusion, use separated members for those things, this could
+	 * be done without breaking network retro-compatibility */
```

Comment:
It's possible but complicated since the DB expects the flags as a single uint32_t. So you'd need to pack them together when storing and unpack the when reading, but then you cannot ensure that flags will be kept in the allowed range when developing new services.

Reply:
In that case when we change this at some point, we should change it in the database too, and probably write some migration code, to migrate from old DB format to the new one